src/test/java/com/marizueva/laboratory/hw2/ex2/LoginTestWithSoftAssertions.java

```diff
@@ -0,0 +1,243 @@
+        softAssert.assertTrue(subMenuFromHeaderUserTableItem.isDisplayed());
+        WebElement subMenuFromHeaderTableWithPagesItem = driver.findElement(By
+                .xpath("//header//ul/li/a[@class='dropdown-toggle']"
+                        + "/following-sibling::ul[@class='dropdown-menu']/li"
+                        + "/a[@href='table-pages.html']"));
+        softAssert.assertTrue(subMenuFromHeaderTableWithPagesItem.isDisplayed());
+        WebElement subMenuFromHeaderDifferentElementsItem = driver.findElement(
+                By.xpath("//header//ul/li/a[@class='dropdown-toggle']"
+                        + "/following-sibling::ul[@class='dropdown-menu']"
+                        + "/li/a[@href='different-elements.html']"));
+        softAssert.assertTrue(subMenuFromHeaderDifferentElementsItem.isDisplayed());
+
+        // Click on Service subcategory in the left section and check that
+        // drop down contains options
+        //"Support, Dates, Complex Table, Simple Table, Tables With Pages,
+        // Different Elements"
+        WebElement leftServiceMenu = driver.findElement(By
+                .cssSelector("ul.sidebar-menu.left >li[index='3']"));
+        leftServiceMenu.click();
+
+        WebElement subMenuFromLeftSupportItem = driver.findElement(
+                By.xpath("//div[@name='navigation-sidebar']//span[text()='Support']"));
+        softAssert.assertTrue(subMenuFromLeftSupportItem.isDisplayed());
+        WebElement subMenuFromLeftDatesItem = driver.findElement(
+                By.xpath("//div[@name='navigation-sidebar']//span[text()='Dates']"));
+        softAssert.assertTrue(subMenuFromLeftDatesItem.isDisplayed());
+        WebElement subMenuFromLeftSearchItem = driver.findElement(
+                By.xpath("//div[@name='navigation-sidebar']//span[text()='Search']"));
+        softAssert.assertTrue(subMenuFromLeftSearchItem.isDisplayed());
+        WebElement subMenuFromLeftComplexTableItem = driver.findElement(
+                By.xpath("//div[@name='navigation-sidebar']//a[@href='complex-table.html']"));
+        softAssert.assertTrue(subMenuFromLeftComplexTableItem.isDisplayed());
+        WebElement subMenuFromLeftUserTableItem = driver.findElement(
+                By.xpath("//div[@name='navigation-sidebar']//a[@href='user-table.html']"));
+        softAssert.assertTrue(subMenuFromLeftUserTableItem.isDisplayed());
+        WebElement subMenuFromLeftTableWithPagesItem = driver.findElement(
+                By.xpath("//div[@name='navigation-sidebar']"
+                        + "//a[@href='table-pages.html']"));
+        softAssert.assertTrue(subMenuFromLeftTableWithPagesItem.isDisplayed());
+        WebElement subMenuFromLeftDifferentElementsItem = driver.findElement(
+                By.xpath("//div[@name='navigation-sidebar']"
+                        + "//a[@href='different-elements.html']"));
+        softAssert.assertTrue(subMenuFromLeftDifferentElementsItem.isDisplayed());
```

Comment:
Looks like, that here you could get `List<WebElement>` for the menu items and thought the loop check that all they displayed instead of getting each element from the page. 
It increases speed of your test

Reply:
fixed